packages/xo-server/src/index.js

```diff
@@ -278,14 +279,21 @@ async function setUpPassport(express, xo, { authentication: authCfg }) {
 
   // Install the local strategy.
   xo.registerPassportStrategy(
-    new LocalStrategy(async (username, password, done) => {
-      try {
-        const { user } = await xo.authenticateUser({ username, password })
-        done(null, user)
-      } catch (error) {
-        done(null, false, { message: error.message })
+    new LocalStrategy(
+      { passReqToCallback: true },
+      async (req, username, password, done) => {
+        const zone = Zone.current.fork('webAuthentication')
+        zone.data.userIp = req.socket.remoteAddress
```

Comment:
Why using `Zone` instead of passing `userIp` directly?

Reply:
To not have to passe it throughout the chain call.

Reply:
Why??

Reply:
Because it's only used on the authentication without token and it's a client data which can be used on all the chain call level when needed.